### DISCUSSION TITLE
Fief causes numerous idle sessions in PostgreSQL

### DISCUSSION THREAD

**Discussion by meteoDaniel** [fief-dev/fief#383] [Q&A]
## Describe the bug

In my database, I can see numerous idle sessions that caused an outage. No more clients were allowed in my PostgreSQL 

## To Reproduce

![Bildschirmfoto vom 2024-05-29 22-08-59](https://github.com/fief-dev/fief/assets/27021858/45205c18-369f-43fd-a032-7a4c1b511cba)

## Expected behavior

Sessions will be closed correctly

## Configuration

* If self-hosted, Fief version: latest

## Additional context

I recommend to use a contextmanager to handle database sessions: 

```python
@contextmanager
def get_scoped_session(database: str) -> Generator[Session, None, None]:
    """
    Returns a scoped sqlalchemy session for the specified database

    Args:
        database: database name

    Returns:
        scoped session object

    """
    session = sessionmaker(bind=create_engine(database))()

    try:
        yield session
        session.commit()
    except Exception:
        session.rollback()
        raise
    finally:
        session.close()

# Usage: 
with get_scoped_session(engine):
    # do stuff 

```

**Comment by frankie567** [comment]
We do use a context manager to manage session lifetime:

https://github.com/fief-dev/fief/blob/9c7e72e4d37c078c722a5445bc12a6c1e159b274/fief/dependencies/db.py#L9-L19

It's done implicitly by FastAPI, but it's there.

What's happening however is that SQLAlchemy maintains [connection pools](https://docs.sqlalchemy.org/en/20/core/pooling.html) (i.e. idle opened connection ready to answer incoming queries). The default of SQLAlchemy is 5 opened connections at most. The thing is, they can pile up quite quickly if you have several server processes. We don't have a setting right now to tweak the number of connections, but it should probably be added.

Meanwhile, I suggest you to reduce the number of processes of the server: set the `WEB_CONCURRENCY` variable to `1` or `2`.

**Reply by meteoDaniel** [reply]
Dear @frankie567 , thanks for the Support :) 

I will adapt the config. 

Best regards

**Reply by frankie567** [reply]
Tracking the connection pool setting addition here: #386
